TestIsAddressCovered() fails for localhost on Fedora/CentOS

# Required information

 * Distribution: Fedora 30/31, CentOS 7/8

# Issue description

Since LXD >=3.19 the tests for `github.com/lxc/lxd/lxd/util` fail with the error:
```
+ go test -buildmode pie -compiler gc -v -tags libsqlite3 -ldflags ' -extldflags '\''-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '\''' github.com/lxc/lxd/lxd/util
=== RUN   Test_CompareConfigsMismatch                  
=== RUN   Test_CompareConfigsMismatch/different_values_for_keys:_foo                                                   
=== RUN   Test_CompareConfigsMismatch/different_values_for_keys:_egg,_foo
--- PASS: Test_CompareConfigsMismatch (0.00s)   
    --- PASS: Test_CompareConfigsMismatch/different_values_for_keys:_foo (0.00s)
    --- PASS: Test_CompareConfigsMismatch/different_values_for_keys:_egg,_foo (0.00s)
=== RUN   Test_CompareConfigs                                                                                          
--- PASS: Test_CompareConfigs (0.00s) 
=== RUN   TestInMemoryNetwork                                                                                          
--- PASS: TestInMemoryNetwork (0.00s)                                                                                  
=== RUN   TestCanonicalNetworkAddress                                                                                  
=== RUN   TestCanonicalNetworkAddress/foo.bar                                                                          
=== RUN   TestCanonicalNetworkAddress/192.168.1.1:443                                                                  
=== RUN   TestCanonicalNetworkAddress/f921:7358:4510:3fce:ac2e:844:2a35:54e                    
=== RUN   TestCanonicalNetworkAddress/127.0.0.1                                                                        
--- PASS: TestCanonicalNetworkAddress (0.00s)                                                                          
    --- PASS: TestCanonicalNetworkAddress/foo.bar (0.00s)                                                                                                                                                                                     
    --- PASS: TestCanonicalNetworkAddress/192.168.1.1:443 (0.00s)                                                                                                                                                                             
    --- PASS: TestCanonicalNetworkAddress/f921:7358:4510:3fce:ac2e:844:2a35:54e (0.00s)                                
    --- PASS: TestCanonicalNetworkAddress/127.0.0.1 (0.00s)      
=== RUN   TestIsAddressCovered                                                                                         
=== RUN   TestIsAddressCovered/127.0.0.1:8443-127.0.0.1:8443
=== RUN   TestIsAddressCovered/garbage-127.0.0.1:8443
=== RUN   TestIsAddressCovered/127.0.0.1:8444-garbage
=== RUN   TestIsAddressCovered/127.0.0.1:8444-127.0.0.1:8443
=== RUN   TestIsAddressCovered/127.0.0.1:8443-0.0.0.0:8443                                                             
=== RUN   TestIsAddressCovered/[::1]:8443-0.0.0.0:8443
=== RUN   TestIsAddressCovered/:8443-0.0.0.0:8443                                                                                                                                                                                             
=== RUN   TestIsAddressCovered/127.0.0.1:8443-[::]:8443
=== RUN   TestIsAddressCovered/[::1]:8443-[::]:8443                                                                    
=== RUN   TestIsAddressCovered/[::1]:8443-:8443                                                                        
=== RUN   TestIsAddressCovered/:8443-[::]:8443  
=== RUN   TestIsAddressCovered/0.0.0.0:8443-[::]:8443                                                                  
=== RUN   TestIsAddressCovered/127.0.0.1:8443-localhost:8443                         
=== RUN   TestIsAddressCovered/[::1]:8443-ip6-localhost:8443                                                           
--- FAIL: TestIsAddressCovered (0.00s)
    --- PASS: TestIsAddressCovered/127.0.0.1:8443-127.0.0.1:8443 (0.00s)                                               
    --- PASS: TestIsAddressCovered/garbage-127.0.0.1:8443 (0.00s)                                                      
    --- PASS: TestIsAddressCovered/127.0.0.1:8444-garbage (0.00s)                                                      
    --- PASS: TestIsAddressCovered/127.0.0.1:8444-127.0.0.1:8443 (0.00s)                                               
    --- PASS: TestIsAddressCovered/127.0.0.1:8443-0.0.0.0:8443 (0.00s)                                                 
    --- PASS: TestIsAddressCovered/[::1]:8443-0.0.0.0:8443 (0.00s)                             
    --- PASS: TestIsAddressCovered/:8443-0.0.0.0:8443 (0.00s)                                                          
    --- PASS: TestIsAddressCovered/127.0.0.1:8443-[::]:8443 (0.00s)                                                    
    --- PASS: TestIsAddressCovered/[::1]:8443-[::]:8443 (0.00s)                                                                                                                                                                               
    --- PASS: TestIsAddressCovered/[::1]:8443-:8443 (0.00s)                                                                                                                                                                                   
    --- PASS: TestIsAddressCovered/:8443-[::]:8443 (0.00s)                                                             
    --- PASS: TestIsAddressCovered/0.0.0.0:8443-[::]:8443 (0.00s)
    --- FAIL: TestIsAddressCovered/127.0.0.1:8443-localhost:8443 (0.00s)
        net_test.go:80:                              
                Error Trace:    net_test.go:80       
                Error:          Should be true 
                Test:           TestIsAddressCovered/127.0.0.1:8443-localhost:8443                                     
    --- FAIL: TestIsAddressCovered/[::1]:8443-ip6-localhost:8443 (0.00s)
        net_test.go:80:                                                                                                                                                                                                                       
                Error Trace:    net_test.go:80         
                Error:          Should be true                                                                         
                Test:           TestIsAddressCovered/[::1]:8443-ip6-localhost:8443                                     
=== RUN   TestListenImplicitIPv6Wildcard        
--- PASS: TestListenImplicitIPv6Wildcard (0.00s)                                                                       
FAIL                                                                                                                   
FAIL    github.com/lxc/lxd/lxd/util     0.009s
```

The reason for the first error is that `localhost` is resolved to `::1` instead of the expected `127.0.0.1`:
```
[mockbuild@fb2a1cf064014f39b8a3d6023ed0e819 ~]$ cat /etc/hosts
127.0.0.1 localhost localhost.localdomain
::1       localhost localhost.localdomain localhost6 localhost6.localdomain6
```
```
[mockbuild@fb2a1cf064014f39b8a3d6023ed0e819 ~]$ getent hosts localhost
::1             localhost localhost.localdomain localhost6 localhost6.localdomain6
```
The second error is because `ip6-localhost` is not defined.

So far I simply remove those tests from being run when building the RPM ([lxd-3.19-util-net-Disable-testing-of-localhost.patch](https://github.com/ganto/copr-lxc3/blob/master/lxd/lxd-3.19-util-net-Disable-testing-of-localhost.patch)) but maybe there is a better solution?

--- a/lxd/util/net_test.go
+++ b/lxd/util/net_test.go
@@ -52,11 +52,13 @@ func TestCanonicalNetworkAddress(t *testing.T) {
 }
 
 func TestIsAddressCovered(t *testing.T) {
-	cases := []struct {
+	type testCase struct {
 		address1 string
 		address2 string
 		covered  bool
-	}{
+	}
+
+	cases := []testCase{
 		{"127.0.0.1:8443", "127.0.0.1:8443", true},
 		{"garbage", "127.0.0.1:8443", false},
 		{"127.0.0.1:8444", "garbage", false},
@@ -69,8 +71,17 @@ func TestIsAddressCovered(t *testing.T) {
 		{"[::1]:8443", ":8443", true},
 		{":8443", "[::]:8443", true},
 		{"0.0.0.0:8443", "[::]:8443", true},
-		{"127.0.0.1:8443", "localhost:8443", true},
-		{"[::1]:8443", "ip6-localhost:8443", true},
+	}
+
+	// Test some localhost cases too
+	ips, err := net.LookupHost("localhost")
+	if err == nil && len(ips) > 0 && ips[0] == "127.0.0.1" {
+		cases = append(cases, testCase{"127.0.0.1:8443", "localhost:8443", true})
+	}
+
+	ips, err = net.LookupHost("ip6-localhost")
+	if err == nil && len(ips) > 0 && ips[0] == "::1" {
+		cases = append(cases, testCase{"[::1]:8443", "ip6-localhost:8443", true})
 	}
 
 	for _, c := range cases {

